pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertTTLConfig.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+
+/**
+ * The {@code UpsertTTLConfig} class contains the upsert TTL (time-to-live) related configurations.
+ * Pinot upsert keeps track of all primary keys in heap, it's costly and also affects performance when table is large.
+ *
+ * If primary keys in the table have lifecycle, they won't get updated after a certain period time, then we can use the
+ * following configuration to enable upsert ttl feature. Pinot will only keeps track of alive primary keys in heap.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class UpsertTTLConfig {
```

Comment:
To make it more general, shall we consider using an absolute long (or double) value as the TTL? In order to enable TTL, the comparison column must be a number type (we should also verify that when validating the table config), and we use absolute value to determine whether the key is expired. E.g. the latest comparison value we got is 10000, and TTL is set to 1000, then any comparison value smaller than 9000 are expired, and we put 9000 as the watermark.

Reply:
@Jackie-Jiang can you please confirm if you want me to make the above changes?